### ISSUE TITLE
Loading large web content leads to blank screen on iOS physical device

### ISSUE THREAD

**Issue by dev-nicelee** [pichillilorenzo/flutter_inappwebview#1709]
- [x] I have read the [Getting Started](https://inappwebview.dev/docs/intro/) section
- [x] I have already searched for the same problem

## Environment

| Technology      | Version       |
|-----------------| ------------- |
| Flutter version |      3.10.4         |
| Plugin version  |      5.7.2+3         |
| Android version |               |
| iOS version     |       16.5.1        |
| macOS version   |               |
| Xcode version   |       14.1        |

Device information: 

## Description

**Expected behavior:** 

Shows html content on screen.

**Current behavior:** 

I have inappwebview widget that resolves the web content including html string and link. For some content (usually large content with images), sometimes it crashes and shows a blank screen on `physical ios device`

## Steps to reproduce

1. Create inappwebview widget which contains the logic to calculate the scroll height of the content. (the example code below)
2. Set the large web content for the webview (example link: https://medium.com/@ahmad.hamoush.785/8-flutter-widgets-every-app-should-contain-5c6129a6928b)
3. Go to the webview page
4. The web content is blipped then disappeared soon.

This is the build method which renders the web content via inappwebview.

```
@override
  Widget build(BuildContext context) {
    return Container(
      // This is 
      height: contentHeight + 80,
      padding: const EdgeInsets.fromLTRB(22, 16, 22, 20),
      child: Stack(
        children: [
          if ((widget.htmlContent != null || widget.htmlUri != null) &&
              widget.isContentLoaded &&
              !hideWebView)
            Stack(
              children: [
                InAppWebView(
                  key: widget.key,
                  initialUrlRequest: URLRequest(
                       url: Uri.parse(
                           'https://medium.com/@ahmad.hamoush.785/8-flutter-widgets-every-app-should-contain-5c6129a6928b')),
                  initialOptions: InAppWebViewGroupOptions(
                    crossPlatform: InAppWebViewOptions(
                        javaScriptEnabled: true,
                        verticalScrollBarEnabled: false,
                        transparentBackground: true,
                        useShouldOverrideUrlLoading: true),
                    android: AndroidInAppWebViewOptions(
                      useHybridComposition: true,
                    ),
                    ios: IOSInAppWebViewOptions(
                      allowsInlineMediaPlayback: true,
                      allowsBackForwardNavigationGestures: true,
                      allowsLinkPreview: true,
                    ),
                  ),
                  onWebViewCreated: _onWebViewCreated,
                  onLoadStop: _onLoadStop,
                ),
                if (widget.onTapWebview != null)
                  GestureDetector(
                    onTap: widget.onTapWebview,
                  ),
              ],
            ),

          widget.isContentLoaded
              ? const SizedBox.shrink()
              : Container(
                  color: AppColors.baseLight[900],
                )
        ],
      ),
    );
```

onLoadStop callback which calculates the scroll height of the content

```
void _onLoadStop(InAppWebViewController webViewController, Uri? url) {
  webViewController
      .evaluateJavascript(source: 'document.body.scrollHeight;')
      .then((value) {
    setState(() {
      contentHeight = double.parse(value.toString());
    });
  });
}
```

## Images   

https://github.com/pichillilorenzo/flutter_inappwebview/assets/32350596/0f88ddc9-7bce-4c64-b6f7-3ed448471063

## Stacktrace/Logcat   

```
[Security] __delegate_identifier__:Performance Diagnostics__:::____message__:This method should not be called on the main thread as it may lead to UI unresponsiveness.
[Process] 0x118000a20 - [PID=34754] WebProcessProxy::didClose: (web process 0 crash)
[Process] 0x118000a20 - [PID=34754] WebProcessProxy::processDidTerminateOrFailedToLaunch: reason=Crash
[ProcessSuspension] 0x11701c5a0 - ProcessAssertion: Failed to acquire RBS Background assertion 'XPCConnectionTerminationWatchdog' for process because PID 0 is invalid
[ProcessSuspension] 0x11701c5a0 - ProcessAssertion::acquireSync Failed to acquire RBS assertion 'XPCConnectionTerminationWatchdog' for process with PID=0, error: (null)
[Process] 0x10567b018 - [pageProxyID=26, webPageID=27, PID=34754] WebPageProxy::processDidTerminate: (pid 34754), reason=Crash
[Loading] 0x10567b018 - [pageProxyID=26, webPageID=27, PID=34754] WebPageProxy::dispatchProcessDidTerminate: reason=Crash
[Process] 0x117078380 - GPUProcessProxy::didClose:
[Process] 0x117078380 - GPUProcessProxy::gpuProcessExited: reason=Crash
[assertion] Error acquiring assertion: <Error Domain=RBSServiceErrorDomain Code=1 "target is not running or doesn't have entitlement com.apple.runningboard.assertions.webkit" UserInfo={NSLocalizedFailureReason=target is not running or doesn't have entitlement com.apple.runningboard.assertions.webkit}>
[ProcessSuspension] 0x11701c120 - ProcessAssertion::acquireSync Failed to acquire RBS assertion 'GPUProcess Background Assertion' for process with PID=34755, error: Error Domain=RBSServiceErrorDomain Code=1 "target is not running or doesn't have entitlement com.apple.runningboard.assertions.webkit" UserInfo={NSLocalizedFailureReason=target is not running or doesn't have entitlement com.apple.runningboard.assertions.webkit}
2
Could not signal service com.apple.WebKit.WebContent: 113: Could not find specified service
```

**Comment by ahmetbesli** [comment]
@dev-nicelee I have the same issue now. I couldn't find any solution until now. Using different lib, or different type of browser will be fine but I don't want to change whole structure. I am not able to test it on

**iOS version | 16.5.1**

But there is no problem with the 16.1, Apple is just declining my app in review because of this failure. Hope to see it fixed soon

**Comment by cooaer** [comment]
Same Issue. No solution!
```
[Process] 0x12a013620 - [PID=18746] WebProcessProxy::didClose: (web process 0 crash)
[Process] 0x12a013620 - [PID=18746] WebProcessProxy::processDidTerminateOrFailedToLaunch: reason=Crash
[ProcessSuspension] 0x12801cde0 - ProcessAssertion: Failed to acquire RBS Background assertion 'XPCConnectionTerminationWatchdog' for process because PID 0 is invalid
[ProcessSuspension] 0x12801cde0 - ProcessAssertion::acquireSync Failed to acquire RBS assertion 'XPCConnectionTerminationWatchdog' for process with PID=0, error: (null)
[Process] 0x10ba6e018 - [pageProxyID=154, webPageID=155, PID=18746] WebPageProxy::processDidTerminate: (pid 18746), reason=Crash
```

**Comment by tony-Lei** [comment]
+1

Any solution?

**Comment by dev-nicelee** [comment]
`tl;dr`: Letting the webview render the entire screen without calculating the height dynamically.

For those of you have stuck on this issue.

In my team, we've changed the package to the official [webview_flutter](https://pub.dev/packages/webview_flutter) and given up the dynamic height of the content view since it's ongoing issue and it is unlikely to solve in the near future.

What we did was replacing entire screen with the webview and allowing the scrolling down the whole content except parts which an app can handle while it doesn't hurt the ui (e.g menu header, modal etc)

I hope it helps to your future development and I'm closing this issue.

You guys can reopen the issue whenever it should.🤞
